main.tf

```diff
@@ -3,7 +3,10 @@ data "aws_caller_identity" "current" {}
 data "aws_region" "current" {}
 
 resource "aws_glue_catalog_database" "this" {
-  name        = var.name
+  // dashes in glue table names may cause errors when running Athena DDL queries.
+  // cf. https://aws.amazon.com/premiumsupport/knowledge-center/parse-exception-missing-eof-athena/
+  name = replace(var.name, "-", "_")
```

Comment:
instead of "silently" modifying the passed `var.name`, I'd recommend to add variable validation (https://www.terraform.io/docs/language/values/variables.html#custom-validation-rules) on `var.name`.

Reply:
@jansiwy The variable `name` is used for s3 bucket and glue catalog database name. The s3 buckets cannot contain underscores, the glue db names seem to produces errors (sometimes) for any special character other than underscores.

Would you suggest passing in `glue_catalog_database_name` and `bucket_name` and `athena_workspace_name` (also based on `var.name`) separately?

Reply:
hmm... ok, let's keep the `replace` function as is.